app/api/currentuser/withdraw_group_leave_request.go

```diff
@@ -0,0 +1,46 @@
+package currentuser
+
+import (
+	"net/http"
+
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+// swagger:operation DELETE /current-user/group-leave-requests/{group_id} groups users groupLeaveRequestWithdraw
+// ---
+// summary: Withdraw a request to leave a group
+// description: >
+//   Lets a user withdraw a request to leave a group.
+//
+//     On success the service removes a row  with `group_id` = `{group_id}`,
+//     `type` = 'leave_request' and `member_id` = user's `group_id` from the `group_pending_requests` table
+//     and creates a new row in `group_membership_changes`
+//     with `action` = `leave_request_withdrawn` and `at` equal to current UTC time.
+//
+//
+//   The user should be a member of the group and there should be a row with
```

Comment:
Maybe we shouldn't check that the user "should be a member of the group", it is not important. As it is about himself, he should be able to remove it anyway.

Reply:
it's impossible to have a pending leave request while not being a group member

Reply:
Still the same topic: it is "impossible" through the current services with no bug, but what it is the case in the db? It does not cost us much more to write it in a way which does not block.

Reply:
I think we still should check this since the leave requests without active group membership are not something that can be accepted. Instead of allowing to accept them, we will remove them on doing other transitions.